File: src/main/java/org/springframework/samples/petclinic/system/Page.java
package org.springframework.samples.petclinic.system;

import java.util.Iterator;
import java.util.List;

public class Page<T> {
	private final List<T> values;
	private final int totalPages;

	public Page(List<T> values, Pageable pageable) {
		this.values = values;
		this.totalPages = 1;
	}

	public Page(List<T> values) {
		this.values = values;
		this.totalPages = 1;
	}

	public List<T> toList() {
		return values;
	}

	public List<T> getContent() {
		return values;
	}

	public int getTotalPages() {
		return totalPages;
	}

	public int getTotalElements() {
		return values.size();
	}

	public boolean isEmpty() {
		return values.isEmpty();
	}

	public Iterator<T> iterator() {
		return values.iterator();
	}
}

File: src/main/kotlin/org/springframework/samples/petclinic/system/Page.kt
package org.springframework.samples.petclinic.system

class Page<T>(values: List<T>, pageable: Pageable? = null) {
    val content: List<T> = values
    val totalPages: Int = 1

    fun toList(): List<T> = content

    val totalElements: Int get() = content.size

    val isEmpty: Boolean get() = content.isEmpty()

    fun iterator(): Iterator<T> = content.iterator()
}

object Pageable {
	fun unpaged(): Pageable? = null
}

object PageRequest {
	fun of(i: Int, pageSize: Int): Pageable? = null
}
